spec/graphql/query_spec.rb

```diff
@@ -972,4 +972,23 @@ def self.after_multiplex(m)
       assert_equal "order_by", order_by_argument_value.definition.graphql_name
     end
   end
+
+  describe "when provided input object field names are not unique" do
+    let(:variables) { {} }
+    let(:result) { Dummy::Schema.execute(query_string, variables: variables) }
+
+    describe "the query is invalid" do
+      let(:query_string) {%|
+        query getCheeses{
+          searchDairy(product: [{ source: COW, source: COW }]) {
+            __typename
+          }
+        }
+      |}
+
+      it "returns errors" do
+        refute_nil(result["errors"])
```

Comment:
This test was the one I added originally to confirm the issue. Do you want this improved, moved, or deleted if the validation-specific spec below is sufficient?